kube-derive/src/custom_resource.rs

```diff
@@ -1,496 +1,372 @@
-use crate::{CustomDerive, ResultExt};
+use darling::FromDeriveInput;
 use inflector::string::pluralize::to_plural;
 use proc_macro2::{Ident, Span};
-use syn::{Data, DeriveInput, Path, Result, Visibility};
-
-#[derive(Debug)]
-pub(crate) struct CustomResource {
-    tokens: proc_macro2::TokenStream,
-    ident: proc_macro2::Ident,
-    visibility: Visibility,
-    kubeattrs: KubeAttrs,
-}
+use syn::{Data, DeriveInput, Path};
 
 /// Values we can parse from #[kube(attrs)]
-#[derive(Debug, Default)]
+#[derive(Debug, Default, FromDeriveInput)]
+#[darling(attributes(kube))]
 struct KubeAttrs {
```

Comment:
This is awesome. Gets rid of so much code, and darling seems pretty light on dependencies as well. Great clean-up.